components/infobox/wikis/mobilelegends/infobox_league

```diff
@@ -0,0 +1,339 @@
+		return nil
+	end
+
+	if localCurrency then
+		localCurrency = string.upper(localCurrency)
+		local exchangeDate = Variables.varDefault('tournament_enddate', _TODAY)
+		local exchangeRate = CustomLeague:_currencyConversion(
+			1,
+			localCurrency,
+			exchangeDate
+		)
+		--set currency vars for usage in prize pools
+		Variables.varDefine('tournament_currency_rate', exchangeRate or '')
+		Variables.varDefine('tournament_currency', localCurrency)
+		if prizePool and not prizePoolUSD then
+			if not exchangeRate then
+				error('Invalid local currency "' .. localCurrency .. '"')
+			end
+			prizePoolUSD = exchangeRate * prizePool
+		end
+	end
+
+	Variables.varDefine('tournament_prizepoolusd', prizePoolUSD or prizePool)
+	Variables.varDefine('tournament_prizepoollocal', prizePool)
+
+	if prizePoolUSD and prizePool then
+		return Template.safeExpand(
+			mw.getCurrentFrame(),
+			'Local currency',
+			{(localCurrency or 'usd'):lower(), prizepool = CustomLeague:_displayPrizeValue(prizePool, 2)}
+		) .. '<br>(≃ $' .. CustomLeague:_displayPrizeValue(prizePoolUSD) .. ' ' .. _ABBR_USD .. ')'
+	elseif prizePoolUSD then
+		return '$' .. CustomLeague:_displayPrizeValue(prizePoolUSD, 2) .. ' ' .. _ABBR_USD
+	elseif prizePool then
+		return Template.safeExpand(
+			mw.getCurrentFrame(),
+			'Local currency',
+			{(localCurrency or 'usd'):lower(), prizepool = CustomLeague:_displayPrizeValue(prizePool, 2)}
+		)
+	end
+end
+
+function CustomLeague:_createTierDisplay()
+	local tier = _args.liquipediatier or ''
+	local tierType = _args.liquipediatiertype or _args.tiertype or ''
+	if String.isEmpty(tier) then
+		return nil
+	end
+
+	local tierText = Tier['text'][tier]
+	local hasInvalidTier = tierText == nil
+	tierText = tierText or tier
+
+	local hasInvalidTierType = false
+
+	local output = '[[' .. tierText .. ' Tournaments|' .. tierText .. ']]'
+		.. '[[Category:' .. tierText .. ' Tournaments]]'
+
+	if not String.isEmpty(tierType) then
+		tierType = Tier['types'][string.lower(tierType or '')] or tierType
+		hasInvalidTierType = Tier['types'][string.lower(tierType or '')] == nil
+		tierType = '[[' .. tierType .. ' Tournaments|' .. tierType .. ']]'
+			.. '[[Category:' .. tierType .. ' Tournaments]]'
+		output = tierType .. '&nbsp;(' .. output .. ')'
+	end
+
+	output = output ..
+		(hasInvalidTier and '[[Category:Pages with invalid Tier]]' or '') ..
+		(hasInvalidTierType and '[[Category:Pages with invalid Tiertype]]' or '')
+
+	Variables.varDefine('tournament_tier', tier)
+	Variables.varDefine('tournament_tiertype', tierType)
+	return output
+end
+
+function CustomLeague._getPatchVersion()
+	if String.isEmpty(_args.patch) then return nil end
+	
+	local content = PageLink.makeInternalLink(_args.patch, 'Patch ' .. _args.patch)
+	
+	if not String.isEmpty(_args.epatch) then
+		content = content .. '&nbsp;&ndash;&nbsp;'
+		content = content .. PageLink.makeInternalLink(_args.epatch, 'Patch ' .. _args.epatch)
+	end
+
+	return content
+end
```

Comment:
Maybe create a default handling on Commons for patch? There is for lpdb storage, but not display

Reply:
Same to Prize and Tier, it is a worth commons-managed.